### PR TITLE
Registry protocol listener

### PR DESCRIPTION
**Pull request by beiwei30**
## What is the purpose of the change

XXXXX

## Brief changelog

XXXXX

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ConsumerModel.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 import org.apache.dubbo.common.utils.Assert;
 import org.apache.dubbo.config.ReferenceConfigBase;
+import org.apache.dubbo.config.annotation.Service;
```
Comment (lzf971107) [inline dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ConsumerModel.java]: Please delete unused import, otherwise you can't pass ci check.

Comment (beiwei30) [reply dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ConsumerModel.java]: thanks, will do.

File: dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ServiceRepository.java [hunk 2]
```diff
@@ -84,33 +90,39 @@ public void registerConsumer(String serviceKey,
...
+                serviceMetadata);
+        providers.putIfAbsent(serviceKey, providerModel);
+        providersWithoutGroup.putIfAbsent(keyWithoutGroup(serviceKey), providerModel);
+    }
+
+    private static String keyWithoutGroup(String serviceKey) {
+        return interfaceFromServiceKey(serviceKey) + ":" + versionFromServiceKey(serviceKey);
+    }
+
+    public void reRegisterProvider(String newServiceKey, String serviceKey) {
```
Comment (qinliujie) [inline dubbo-common/src/main/java/org/apache/dubbo/rpc/model/ServiceRepository.java]: if newServiceKey equals serviceKey,then ProviderModel would be remove

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryDirectory.java [hunk 3]
```diff
@@ -616,8 +641,14 @@ public URL getRegisteredConsumerUrl() {
         return registeredConsumerUrl;
     }
 
-    public void setRegisteredConsumerUrl(URL registeredConsumerUrl) {
-        this.registeredConsumerUrl = registeredConsumerUrl;
+    public void setRegisteredConsumerUrl(URL url) {
+        if (!shouldSimplified) {
```
Comment (qinliujie) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryDirectory.java]: is it should be `if (shouldSimplified) `

**Pull request merged**